py/bitbox02/bitbox02/bitbox02/bitbox02.py

```diff
@@ -108,6 +115,19 @@ def device_info(self) -> Dict[str, Any]:
             "monotonic_increments_remaining": response.device_info.monotonic_increments_remaining,
         }
 
+    def check_firmware_version(self) -> bool:
+        """
+        Returns False if the bitbox02 version is not what the library expects.
+        The connection should already bail early in case of a version mismatch,
+        so False indicates that some expected functionality might be missing.
+        """
+        # pylint: disable=no-member
+        _, _, edition, _ = self.get_info()
```

Comment:
get_info (OP_INFO) was added in v4.2.0 i think, so this will fail for earlier versions. It is also a bit strange to get the edition from `get_info()` but `self.version` from the USB descriptor; both should come from the same source.

Please add `self.edition` alongside `self.version` in __init__, based on the descriptor, same as in the Go lib:

https://github.com/digitalbitbox/bitbox02-api-go/blob/master/api/common/common.go#L45-L57

Reply:
Now I am confused, why is there a check_firmware_version and a check_version method?

Reply:
Removed the method here.